git_machete/cmd.py

```diff
@@ -339,359 +339,1027 @@ def popen_git(cli_ctxt: CommandLineContext, git_cmd: str, *args: str, **kwargs:
 
 # Manipulation on definition file/tree of branches
 
-managed_branches: List[str] = []
-down_branches: Dict[str, List[str]] = {}  # TODO (#110): default dict with []
-up_branch: Dict[str, str] = {}  # TODO (#110): default dict with None
-indent: Optional[str] = None
-roots: List[str] = []
-annotations: Dict[str, str] = {}
-
+branch_defs_by_sha_in_reflog: Optional[Dict[str, Optional[List[Tuple[str, str]]]]] = None
```

Comment:
Hmm can this one go into `MacheteClient` as well? 🤔

Reply:
Hmmm TBH as I look on that now... this logic is also super-Machete-specific...

Reply:
I'd pull that (and the associated functions) into the Client as well :/

Reply:
(bump)

Reply:
added as class variable due to usafe in flush_caches funtion